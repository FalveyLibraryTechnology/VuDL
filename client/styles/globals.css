:root,
[color-scheme="light"] {
    --body-bg: #fafbfc;
    --heading-color: #4d4d4d;
    --link-color: #008aed;

    --btn-bg: #485a6c;
    --btn-text: #ffffff;
    --btn-border: #275575;
    --btn-focus-bg: #3498db;

    --btn-primary-bg: #27ae60;
    --btn-primary-text: #ffffff;
    --btn-primary-border: #275575;
    --btn-primary-focus-bg: #155724;

    --btn-danger-bg: red;
    --btn-danger-text: #ffffff;
    --btn-danger-border: #a10705;
    --btn-danger-focus-bg: #a10705;

    --btn-logout-bg: #1abc9c;
    --btn-logout-text: #ffffff;
    --btn-logout-border: #16a085;
    --btn-logout-focus-bg: #1abc9c;

    --btn-toggle-bg: #7f8c8d;
    --btn-toggle-text: #ffffff;
    --btn-toggle-border: #34495e;
    --btn-toggle-focus-bg: #53626f;

    --group-border: #53626f;

    --input-border: #6a737d;
    --input-focus-border: #2188ff;

    --label-hover: #c8e1ff;

    --page-list-bg: #ffffff;
    --page-list-border: #c1c1c1;

    --thumbnail-bg: #e1e1e1;
    --thumbnail-focus-bg: #51aded;
    --thumbnail-number-bg: #ffffff;
    --thumbnail-number-text: #000000;
    --thumbnail-preview-border: #000000;

    --tree-item-focus-bg: #000000;
    --tree-item-focus-text: #ffffff;

    --shadow-gray: #c8e1ff;
}

[color-scheme="dark"] {
}
