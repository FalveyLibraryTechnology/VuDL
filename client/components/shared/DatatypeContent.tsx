import Box from "@mui/material/Box";
import React from "react";
import styles from "./DatatypeContent.module.css";

interface DatatypeContentProps {
    data: string;
    mimeType: string;
}

const DatatypeContent = ({ data, mimeType }: DatatypeContentProps): React.ReactElement => {
    switch (true) {
        case /image\/[-+.\w]+/.test(mimeType):
            return <img className={styles.viewContentImage} src={data} alt="Datastream Image" />;
        case /text\/[-+.\w]+/.test(mimeType):
            return <Box className={styles.viewContentText}>{data}</Box>;
        case /audio\/[-+.\w]+/.test(mimeType):
            return (
                <Box className={styles.viewContentAudio}>
                    <audio controls autoPlay preload="auto">
                        <source src={data} type={mimeType} />
                    </audio>
                </Box>
            );
        case /application\/pdf/.test(mimeType):
        default:
            return <object className={styles.viewContentObject} data={data} type={mimeType} />;
    }
};

export default DatatypeContent;
