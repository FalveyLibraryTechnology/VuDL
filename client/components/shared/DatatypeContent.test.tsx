import React from "react";
import { describe, beforeEach, expect, it } from "@jest/globals";
import renderer from "react-test-renderer";
import DatatypeContent from "./DatatypeContent";

describe("DatatypeContent", () => {
    let props;
    beforeEach(() => {
        props = {
            data: "",
            mimeType: "",
        };
    });

    it("renders img on image primaryType", () => {
        props.data = "test1";
        props.mimeType = "image/jpeg";
        const tree = renderer.create(<DatatypeContent {...props} />);

        expect(tree.toJSON()).toMatchSnapshot();
        expect(tree.root.findByType('img')).toBeTruthy();
    });

    it("renders textarea on text primaryType", () => {
        props.data = "testXml";
        props.mimeType = "text/xml";
        const tree = renderer.create(<DatatypeContent {...props} />);

        expect(tree.toJSON()).toMatchSnapshot();
        expect(tree.root.findByType('div').props.children).toContain(props.data);
    });

    it("renders object on application/pdf", () => {
        props.data = "testPdf";
        props.mimeType = "application/pdf";
        const tree = renderer.create(<DatatypeContent {...props} />);

        expect(tree.toJSON()).toMatchSnapshot();
        expect(tree.root.findByType('object')).toBeTruthy();
    });

    it("renders audio tag on audio primaryType", () => {
        props.data = "testAudio";
        props.mimeType = "audio/mpeg3";
        const tree = renderer.create(<DatatypeContent {...props} />);

        expect(tree.toJSON()).toMatchSnapshot();
        expect(tree.root.findByType('audio')).toBeTruthy();
    });
});
