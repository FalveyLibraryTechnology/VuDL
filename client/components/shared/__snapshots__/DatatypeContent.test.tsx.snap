// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatatypeContent renders audio tag on audio primaryType 1`] = `
<div
  className="viewContentAudio MuiBox-root css-0"
>
  <audio
    autoPlay={true}
    controls={true}
    preload="auto"
  >
    <source
      src="testAudio"
      type="audio/mpeg3"
    />
  </audio>
</div>
`;

exports[`DatatypeContent renders img on image primaryType 1`] = `
<img
  alt="Datastream Image"
  className="viewContentImage"
  src="test1"
/>
`;

exports[`DatatypeContent renders object on application/pdf 1`] = `
<object
  className="viewContentObject"
  data="testPdf"
  type="application/pdf"
/>
`;

exports[`DatatypeContent renders textarea on text primaryType 1`] = `
<div
  className="viewContentText MuiBox-root css-0"
>
  testXml
</div>
`;
